### PR TITLE
feat(injector): inject users and acls

### PR DESCRIPTION
**Pull request by kduret**
## Description

* inject users and acls
* use tagged interator to inject services

**Fixes** MON-20902

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

### CODE REVIEW

File: src/Domain/AclGroup.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace App\Domain;
+
+class AclGroup
+{
+    private ?int $id;
```
Comment (callapa) [inline src/Domain/AclGroup.php:7]: Property must not be accessed before initialization
```suggestion
    private ?int $id = null;
```

File: src/Domain/AclResource.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace App\Domain;
+
+class AclResource
+{
+    private ?int $id;
```
Comment (callapa) [inline src/Domain/AclResource.php:7]: Property must not be accessed before initialization
```suggestion
    private ?int $id = null;
```

File: src/Domain/User.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace App\Domain;
+
+use App\Domain\UserRole;
+
+class User
+{
+    private ?int $id;
```
Comment (callapa) [inline src/Domain/User.php:9]: Property must not be accessed before initialization
```suggestion
    private ?int $id = null;
```

File: src/Infrastructure/MySQL/UserRepository.php [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+                $i . ',' .
+                '"' . $name . $userIndex . '",' .
+                '"' . $alias . $userIndex . '",' .
+                '"' . $name . $userIndex . '@localhost",' .
+                '"en_US.UTF-8",' .
+                '"1",' .
+                '"' . $isAdmin . '",' .
+                '"1"' .
+                '),';
+            if ($insertCount === 50000) {
```
Comment (callapa) [inline src/Infrastructure/MySQL/UserRepository.php:65]: It is safer to base this on the length of the request
`SHOW VARIABLES LIKE 'max_allowed_packet';`

Comment (kduret) [reply src/Infrastructure/MySQL/UserRepository.php:65]: I have created ticket MON-21454 to handle this in all repositories

**Review by callapa — APPROVED** [review]

**Pull request merged**